### PR TITLE
Handle addition of ID to NVT preferences.

### PR DESCRIPTION
**Pull request by kroosec**

### CODE REVIEW

File: src/manage.c [hunk 1]
```diff
@@ -2139,7 +2127,7 @@ send_alive_test_preferences (target_t target)
   if (alive_test == ALIVE_TEST_CONSIDER_ALIVE)
     {
       /* Also select a method, otherwise Ping Host logs a warning. */
-      if (sendf_to_server (OID_PING_HOST ":checkbox:Do a TCP ping <|> yes\n"))
+      if (sendf_to_server (OID_PING_HOST ":1:checkbox:Do a TCP ping <|> yes\n"))
```
Comment (cfi-gb) [inline src/manage.c]: The ids where added to ping_host.nasl today.

File: src/manage.c [hunk 2]
```diff
@@ -5393,7 +5382,7 @@ run_otp_task (task_t task, scanner_t scanner, int from, char **report_id)
                 (gpointer) g_strdup (credential_iterator_private_key
                                       (&credentials)));
 
-              if (sendf_to_server (OID_SSH_AUTH ":file:SSH private key:"
+              if (sendf_to_server (OID_SSH_AUTH ":4:file:SSH private key:"
```
Comment (cfi-gb) [inline src/manage.c]: The ids where added to ssh_authorization_init.nasl today.

File: src/manage.c [hunk 3]
```diff
@@ -5412,9 +5401,9 @@ run_otp_task (task_t task, scanner_t scanner, int from, char **report_id)
           const char *user = credential_iterator_login (&credentials);
           const char *password = credential_iterator_password (&credentials);
 
-          if (sendf_to_server (OID_SMB_AUTH ":entry:SMB login: <|> %s\n",
+          if (sendf_to_server (OID_SMB_AUTH ":1:entry:SMB login: <|> %s\n",
                                user ? user : "")
-              || sendf_to_server (OID_SMB_AUTH ":password:SMB password:"
+              || sendf_to_server (OID_SMB_AUTH ":2:password:SMB password:"
```
Comment (cfi-gb) [inline src/manage.c]: The ids where added to smb_authorization.nasl today.

File: src/manage.c [hunk 4]
```diff
@@ -5443,10 +5432,10 @@ run_otp_task (task_t task, scanner_t scanner, int from, char **report_id)
           const char *user = credential_iterator_login (&credentials);
           const char *password = credential_iterator_password (&credentials);
 
-          if (sendf_to_server (OID_ESXI_AUTH ":entry:ESXi login name:"
+          if (sendf_to_server (OID_ESXI_AUTH ":1:entry:ESXi login name:"
                                " <|> %s\n",
                                user ? user : "")
-              || sendf_to_server (OID_ESXI_AUTH ":password:ESXi login password:"
+              || sendf_to_server (OID_ESXI_AUTH ":2:password:ESXi login password:"
```
Comment (cfi-gb) [inline src/manage.c]: The ids where added to gb_esxi_authorization.nasl today.

File: src/manage.c [hunk 5]
```diff
@@ -5482,25 +5471,25 @@ run_otp_task (task_t task, scanner_t scanner, int from, char **report_id)
...
                                   "SNMPv3 Authentication Algorithm:"
                                   " <|> %s\n",
                                   auth_algorithm ? auth_algorithm : "")
-              || sendf_to_server (OID_SNMP_AUTH ":password:"
+              || sendf_to_server (OID_SNMP_AUTH ":5:password:"
                                   "SNMPv3 Privacy Password:"
                                   " <|> %s\n",
                                   privacy_password ? privacy_password : "")
-              || sendf_to_server (OID_SNMP_AUTH ":radio:"
+              || sendf_to_server (OID_SNMP_AUTH ":6:radio:"
```
Comment (cfi-gb) [inline src/manage.c]: The ids where added to gb_snmp_authorization.nasl today.

File: src/gmp.c [hunk 6]
```diff
@@ -749,8 +752,9 @@ create_config_data_reset (create_config_data_t *data)
     }
 
   g_free (import->preference_alt);
+  g_free (import->preference_id);
   g_free (import->preference_name);
-  g_free (import->preference_name);
+  g_free (import->preference_hr_name);
```
[the remark this answers is not in the document]
Comment (kroosec) [reply src/gmp.c]: Moved to a separate commit.

File: src/gmp.c [hunk 7]
```diff
@@ -3098,23 +3103,24 @@ modify_auth_data_reset (modify_auth_data_t * data)
 static void
 modify_config_data_reset (modify_config_data_t *data)
 {
-  free (data->comment);
-  free (data->config_id);
+  g_free (data->comment);
```
[the remark this answers is not in the document]
Comment (kroosec) [reply src/gmp.c]: Removed the change. I agree that it's better to use a different PR to change all parts together.

File: src/manage.c [hunk 8]
```diff
@@ -2090,42 +2090,34 @@ send_alive_test_preferences (target_t target)
...
-  if (sendf_to_server (OID_PING_HOST ":checkbox:TCP ping tries also TCP-SYN ping"
+  if (sendf_to_server (OID_PING_HOST ":2:checkbox:TCP ping tries also TCP-SYN ping"
                        " <|> %s\n",
                        ((alive_test & ALIVE_TEST_TCP_SYN_SERVICE)
                         && (alive_test & ALIVE_TEST_TCP_ACK_SERVICE))
                         ? "yes"
                         : "no"))
     return -1;
 
-  if (sendf_to_server (OID_PING_HOST ":checkbox:TCP ping tries only TCP-SYN ping"
```
[the remark this answers is not in the document]
Comment (kroosec) [reply src/manage.c]: Fixed. Seemed like a duplicate. Added an ID to it too.

File: src/gmp.c [hunk 9]
```diff
@@ -10120,25 +10132,28 @@ void
...
   default_value = nvt_preference_iterator_value (prefs);
   value = nvt_preference_iterator_config_value (prefs, config);
+  id = nvt_preference_iterator_id (prefs);
 
   if (oid)
     nvt = nvt_name (oid);
   buffer_xml_append_printf (buffer,
                             "<preference>"
                             "<nvt oid=\"%s\"><name>%s</name></nvt>"
+                            "<id>%s</id>"
```
[the remark this answers is not in the document]
Comment (kroosec) [reply src/gmp.c]: Added GMP documentation and compatibility changes section.

**Comment by kroosec** [comment]
> I'm keen to get this in but would like to see the GMP doc updated with the ID, and also a check of why the OID_PING_HOST preference was removed.

Fixed all issues. Also added to the PR a routine to update old-style preferences in config_preferences.

File: src/manage.c [hunk 10]
```diff
@@ -2094,42 +2090,42 @@ send_alive_test_preferences (target_t target)
...
+  if (sendf_to_server (OID_PING_HOST ":7:checkbox:TCP ping tries only TCP-SYN ping"
```
Comment (cfi-gb) [inline src/manage.c]: The id:7 for this preference was added to ping_host.nasl today.

File: src/manage_config_host_discovery.c [hunk 11]
```diff
@@ -122,14 +122,14 @@ check_config_host_discovery (const char *uuid)
...
-               "       AND name = '" OID_GLOBAL_SETTINGS ":checkbox:Strictly unauthenticated';",
+               "       AND name = '" OID_GLOBAL_SETTINGS ":1:checkbox:Strictly unauthenticated';",
                uuid)
       == 0)
     {
       sql ("INSERT INTO config_preferences (config, type, name, value)"
            " VALUES ((SELECT id FROM configs WHERE uuid = '%s'),"
            "         'PLUGINS_PREFS',"
-           "         '" OID_GLOBAL_SETTINGS ":checkbox:Strictly unauthenticated',"
+           "         '" OID_GLOBAL_SETTINGS ":1:checkbox:Strictly unauthenticated',"
```
Comment (cfi-gb) [inline src/manage_config_host_discovery.c]: The id:1 for this preference was added to global_settings.nasl today.

**Pull request merged**